src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -308,5 +308,11 @@ private void AddItemToTrimmedList(string assemblyFilePath, string trimmedAssembl
         var trimmedAssemblyItem = new TaskItem(assemblyFilePath);
         trimmedAssemblyItem.SetMetadata("TrimmedAssemblyFileName", trimmedAssemblyFilePath);
         _trimmedAssemblies.Add(trimmedAssemblyItem);
+
+        // Set the timestamp of trimmed assemblies to be the same as original assemblies.
+        // To avoid undesired re-AOT compilation, during app rebuilding process.
+        FileInfo fileInfo = new FileInfo(assemblyFilePath);
+        DateTime lastModified = fileInfo.LastWriteTime;
+        File.SetLastWriteTime(trimmedAssemblyFilePath, lastModified);
     }
```

Comment:
Still don't think this is the right way, in a build system, build outputs like these should have later timestamps than the build inputs.

Reply:
Without this, AOT generated bitcode files would have older timestamp than the assembly files. Then rebuild will trigger re-AOT, cause it thinks those assembly files have changed.

Reply:
That seems to be a problem with other parts of the build which overwrites build files, its not possible to have correct build dependencies when files are overwritten.

Reply:
TBH, I am not sure why other build files were overwritten but not triggering rebuild. This did fix the test failures tho. Do you have suggestions on fixing it in some other way?

Reply:
Instead of overwriting the original files the item `_WasmAssembliesInternal` can be modified to have the updated paths.

Reply:
Right after the IL strip task, the original assemblies will be replaces with the trimmed ones. The file paths didn't change.

Reply:
I'm suggesting to not replace the original assembly files, and instead just change the paths to new ones in `_WasmAssembliesInternal`. That way rest of the build will be using the newly-trimmed assemblies, instead of the original ones.

Reply:
but if we don't replace the original ones then on a rebuild it should not trigger a rebuild. And this should ensure that we have the correct build dependencies, like Zoltan was talking about.

Reply:
Oh I see what you mean. My worry for this is how we could guarantee tasks run after ILStrip step would always refer to the new ones?

Reply:
`_WasmAssembliesInternal` is used by all the targets, that's the convention used in `WasmApp.Native.targets`, so that would ensure that the new ones are used.

Reply:
We should fix this before merging.

Reply:
Fixed.